lib/AST/USRGeneration.cpp

```diff
@@ -184,8 +191,10 @@ swift::USRGenerationRequest::evaluate(Evaluator &evaluator,
   if (isa<ModuleDecl>(D))
     return std::string(); // Ignore.
 
-  auto interpretAsClangNode = [](const ValueDecl *D)->ClangNode {
-    ClangNode ClangN = D->getClangNode();
+  auto interpretAsClangNode = [&options](const ValueDecl *D) -> ClangNode {
+    auto &ctx = D->getASTContext();
+    auto *importer = static_cast<ClangImporter *>(ctx.getClangModuleLoader());
```

Comment:
Ah sorry I just noticed this, we can't `static_cast` to `ClangImporter` here because libAST does not depend on libClangImporter (which is why it's exposed as ClangModuleLoader). You'll instead need to add `getEffectiveClangNode` as a virtual member to `ClangModuleLoader`.

Reply:
Makes sense. Thanks for explaining.